src/index.js

```diff
@@ -1,390 +1,586 @@
+      } else {
+        element[name][i] = style
       }
-    } else if (node != null && node !== true && node !== false) {
-      children.push(node)
     }
-  }
+  } else {
+    if (name[0] === "o" && name[1] === "n") {
+      name = name.slice(2).toLowerCase()
 
-  return typeof name === "function"
-    ? name(attributes || {}, children)
-    : {
-        nodeName: name,
-        attributes: attributes || {},
-        children: children,
-        key: attributes && attributes.key
+      if (!element.events) element.events = {}
+
+      element.events[name] = nextValue
+
+      if (nextValue == null) {
+        element.removeEventListener(name, eventProxy)
+      } else if (lastValue == null) {
+        element.addEventListener(name, eventProxy)
       }
-}
+    } else {
+      var nullOrFalse = nextValue == null || nextValue === false
 
-export function app(state, actions, view, container) {
-  var map = [].map
-  var rootElement = (container && container.children[0]) || null
-  var oldNode = rootElement && recycleElement(rootElement)
-  var lifecycle = []
-  var skipRender
-  var isRecycling = true
-  var globalState = clone(state)
-  var wiredActions = wireStateToActions([], globalState, clone(actions))
-
-  scheduleRender()
-
-  return wiredActions
-
-  function recycleElement(element) {
-    return {
-      nodeName: element.nodeName.toLowerCase(),
-      attributes: {},
-      children: map.call(element.childNodes, function(element) {
-        return element.nodeType === 3 // Node.TEXT_NODE
-          ? element.nodeValue
-          : recycleElement(element)
-      })
+      if (
+        name in element &&
+        name !== "list" &&
+        name !== "draggable" &&
+        name !== "spellcheck" &&
+        name !== "translate" &&
+        !isSvg
+      ) {
+        element[name] = nextValue == null ? "" : nextValue
+        if (nullOrFalse) {
+          element.removeAttribute(name)
+        }
+      } else {
+        var ns = isSvg && name !== (name = name.replace(/^xlink:?/, ""))
+        if (ns) {
+          if (nullOrFalse) {
+            element.removeAttributeNS(XLINK_NS, name)
+          } else {
+            element.setAttributeNS(XLINK_NS, name, nextValue)
+          }
+        } else {
+          if (nullOrFalse) {
+            element.removeAttribute(name)
+          } else {
+            element.setAttribute(name, nextValue)
+          }
+        }
+      }
     }
   }
+}
 
-  function resolveNode(node) {
-    return typeof node === "function"
-      ? resolveNode(node(globalState, wiredActions))
-      : node != null
-        ? node
-        : ""
+var createElement = function(node, lifecycle, eventProxy, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+        ? document.createElementNS(SVG_NS, node.name)
+        : document.createElement(node.name)
+
+  var props = node.props
+  if (props.onCreate) {
```

Comment:
So lifecycle events are still supported and use camelCase now? 🤔

Reply:
I gave in to camelCase :)

> ...lifecycle events

They're in this branch, for now. I still haven't decided whether to remove them or not. I know we don't need them, but having them can be useful. You can help me decide by adding your feedback to #717.

Reply:
> but having them can be useful.

I'm glad the hooks are still there so I don't yet have to use custom elements and allow `this` in my codebase :nauseated_face: